src/link.rs

```diff
@@ -113,6 +140,42 @@ impl Link {
         return runtime.block_on(Link::get_preview(url));
     }
 
+
+    pub fn load_metadata<P: AsRef<Path>>(
+        &mut self,
+        root: P,
+        resource_id: ResourceId,
+    ) {
+        let metadata_home = root.as_ref().join(METADATA_RELATIVE_PATH);
+        let metadata_path =
+            metadata_home.join(format!("{}", resource_id.crc32));
+
+        if metadata_path.exists() {
+            let metadata_bytes =
+                std::fs::read(metadata_path.to_owned()).unwrap();
+            self.metadata =
+                serde_json::from_slice(metadata_bytes.as_slice()).unwrap()
+        }
+    }
+
+    pub fn save_metadata<P: AsRef<Path>>(
+        &mut self,
+        root: P,
+        metadata: Vec<u8>,
+        resource_id: ResourceId,
+    ) {
+        let metadata_path = root.as_ref().join(METADATA_RELATIVE_PATH);
+        fs::create_dir_all(metadata_path.to_owned())
+            .expect(&format!("Creating {} directory", METADATA_RELATIVE_PATH));
+        let mut metadata_file = File::create(
+            metadata_path
+                .to_owned()
+                .join(format!("{}", resource_id.crc32)),
```

Comment:
Actually, we should also use filesize in the path. Otherwise, in case of a collision by CRC-32 we will write into same file 2 different metadata. CRC-32 + filesize is still not perfect, but reduces frequency of collisions significantly without performance penalty.

The best would be to create a method like `to_filename` printing the whole resource id as a string, for usage in filenames.